example/19_gemm_softmax/gemm_softmax_xdl_fp16.cpp

```diff
@@ -0,0 +1,551 @@
+
+struct SubExp
+{
+    __host__ __device__ constexpr void operator()(EltwiseComputeDataType& dst,
+                                                  const EltwiseComputeDataType& src1,
+                                                  const EltwiseComputeDataType& src2) const
+    {
+        dst = exp(src1 - src2);
+    }
+};
+
+struct Div
+{
+    __host__ __device__ constexpr void operator()(EltwiseComputeDataType& dst,
+                                                  const EltwiseComputeDataType& src1,
+                                                  const EltwiseComputeDataType& src2) const
+    {
+        dst = src1 / src2;
+    }
+};
+
+using DeviceElementwiseSubExpInstance =
+    ck::tensor_operation::device::DeviceBinaryElementwise_2D<CDataType,
+                                                             CDataType,
+                                                             CDataType,
+                                                             EltwiseComputeDataType,
+                                                             SubExp,
+                                                             8>;
+
+using DeviceElementwiseDivInstance =
+    ck::tensor_operation::device::DeviceBinaryElementwise_2D<CDataType,
+                                                             CDataType,
+                                                             CDataType,
+                                                             EltwiseComputeDataType,
+                                                             Div,
+                                                             8>;
+
+using HostGemmInstance = ck::tensor_operation::host::
+    ReferenceGemm<ADataType, BDataType, CDataType, PassThrough, PassThrough, PassThrough>;
+
+using HostReduceMaxInstance = ReductionHost<HostReduceDataType,
+                                            HostReduceDataType,
+                                            HostReduceDataType,
+                                            ReduceMaxId,
+                                            Rank,
+                                            NumReduceDim,
+                                            ReducePropagateNan,
+                                            false>;
+
+using HostReduceSumInstance = ReductionHost<HostReduceDataType,
+                                            AccDataType,
+                                            HostReduceDataType,
+                                            ReduceSumId,
+                                            Rank,
+                                            NumReduceDim,
+                                            ReducePropagateNan,
+                                            false>;
+
+template <typename HostTensorA,
+          typename HostTensorB,
+          typename HostTensorC,
+          typename ComputeDataType,
+          typename Functor,
+          int broadcastDim>
+void host_broadcast2D(
```

Comment:
Put this somewhere in library utilities? Seems like this can be reused in CK profiler

Reply:
I will fix it in another PR https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/178